### DISCUSSION TITLE
Add `wordWrap` function

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#190] [Ideas]
Reference implementation: https://github.com/jonschlinkert/word-wrap

**Comment by MarlonPassos-git** [comment]
I like the idea, and I think the functionality would be a good addition here. However, I see two points that might argue against it:

1. The original implementation is around 60 lines long and seems relatively complex, which could contradict one of Radashi's principles of [Concise Implementations.](https://radashi-org.github.io/our-ethos/#concise-implementations)
2. Even though the other library hasn't been updated for a year, it still seems quite stable. At the moment, I don't see much value in creating our own copy, as the open PRs and issues don't seem particularly relevant.

**Reply by aleclarson** [reply]
1. I don't think we would support every edge case or option that `word-wrap` does.
2. One of the goals of Radashi is to reduce the number of dependencies (and therefore risk of supply chain attacks), so the fact that a function already exists in another package is not a real concern. The discussion in #189 is also relevant here, I think. Our design ethos will naturally lead to a concretely different approach, most of the time.

The biggest concern I have with adding a `wordWrap` function is whether it's actually a popular enough use case. We should probably wait on this one until more demand is clear.

**Reply by kgryte** [reply]
Drive by comment: the reference implementation you linked to in the OP is not a robust implementation. To do it robustly, you need to account for surrogates and grapheme clusters, which would likely require a more complex implementation and potentially additional deps. The `cut` option and its behavior is also too naive. For justified text, using TeX's algorithm would be better.
